tests/test_hdl_ir.py

```diff
@@ -3542,3 +3542,16 @@ class MyEnum(enum.Enum, shape=unsigned(2)):
         self.assertEqual(nl.signal_fields[s4], {
             (): SignalField(nl.signals[s4], signed=False),
         })
+
+
+class CycleTestCase(FHDLTestCase):
+    def test_cycle(self):
+        a = Signal()
+        b = Signal()
+        m = Module()
+        m.d.comb += [
+            a.eq(~b),
+            b.eq(~a),
+        ]
+        with self.assertRaises(CombinationalCycle):
```

Comment:
This should probably assert on a message using `assertRaisesRegex` to skip the variable parts like the line numbers.